examples/hybrid/gravitywave_parameterization/gravity_wave_parameterization.jl

```diff
@@ -0,0 +1,228 @@
+                if (mask[n]) == 1.0
+                    c_hat = c[n] - ᶜu[k]
+                    # f phase speed matches the wind speed, remove c(n) from the set of propagating waves.
+                    if c_hat == 0.0
+                        mask[n] = 0.0
+                    else
+                        # define the criterion which determines if wave is reflected at this level (test).
+                        test = abs(c_hat) * kwv[ink] - ω_r
+                        if test >= 0.0
+                            # wave has undergone total internal reflection. remove it from the propagating set.
+                            mask[n] = 0.0
+                        else
+                            # if wave is not reflected at this level, determine if it is
+                            # breaking at this level (Foc >= 0), or if wave speed relative to
+                            # windspeed has changed sign from its value at the source level
+                            # (c_hat0[n] * c_hat <= 0). if it is above the source level and is
+                            # breaking, then add its momentum flux to the accumulated sum at
+                            # this level.
+                            # set mask=0.0 to remove phase speed band c[n] from the set of active 
+                            # waves moving upwards to the next level.
+                            if c_hat0[n] * c_hat <= 0.0
+                                mask[n] = 0.0
+                                if k > source_level
+                                    fm = fm + B0[n]
+                                end
+                            else
+                                Foc = B0[n] / (c_hat)^3 - fac
+                                if Foc >= 0.0
+                                    mask[n] = 0.0
+                                    if k > source_level
+                                        fm = fm + B0[n]
+                                    end
+                                end
+                            end
+                        end # (test >= 0.0)
+                    end #(c_hat == 0.0)
+                end # mask = 0
+
+            end # phase speed loop
+
+            # TODO: GFDL option to dump remaining flux at the top of the model
+
+            # compute the gravity wave momentum flux forcing 
+            # obtained across the entire wave spectrum at this level.
+            if k > source_level
+                rbh = sqrt(ᶜρ[k] * ᶜρ[k - 1])
+                wave_forcing[k] = (ᶜρ[source_level] / rbh) * fm * eps / ᶜdz[k]
+                wave_forcing[k - 1] = 0.5 * (wave_forcing[k - 1] + wave_forcing[k])
+            else
+                wave_forcing[k] = 0.0
+            end
+
+        end # loop over k
+
+        for k in source_level:length(ᶜu)
+            gwf[k] = gwf[k] + wave_forcing[k]
+        end
+
+    end # loop over ink 
+
+    return reshape(gwf, (length(ᶜu), 1))
```

Comment:
I think we need to figure out a way to avoid `reshape`, here

Reply:
will it work if I initialize `gwf` as `gwf = zeros((length(ᶜu),1))`? @charleskawczynski

Reply:
I think that `reshape` allocates a new array, and `zeros` allocates a new array, too. I think I need to understand the indexing going on in this [line](https://github.com/CliMA/ClimaAtmos.jl/pull/852#discussion_r974836814) to give a better suggestion here